sources/packages/backend/libs/integrations/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt.processing.service.ts

```diff
@@ -124,64 +126,61 @@ export class DisbursementReceiptProcessingService {
               response,
               responseData.header,
               disbursementScheduleId,
-              auditUserId,
+              this.systemUsersService.systemUser.id,
               createdAt,
             );
           if (generatedIdentifier) {
-            result.processSummary.push(
+            processSummary.info(
               `Record with document number ${response.documentNumber} at line ${response.lineNumber} inserted successfully.`,
             );
           } else {
-            result.processSummary.push(
+            processSummary.info(
               `Record with document number ${response.documentNumber} at line ${response.lineNumber} has been ignored as the receipt already exist.`,
             );
           }
         } else {
-          result.processSummary.push(
+          processSummary.info(
             `Document number ${response.documentNumber} at line ${response.lineNumber} not found in SIMS.`,
           );
         }
       } catch (error: unknown) {
-        this.logger.error(error);
         const logMessage = `Unexpected error while processing disbursement receipt record at line ${response.lineNumber}`;
-        result.errorsSummary.push(logMessage);
-        this.logger.error(logMessage);
-        this.logger.error(error);
+        this.logger.error(logMessage, error);
+        processSummary.error(logMessage, error);
+        return;
```

Comment:
Adding the return here will modify the existing behavior. i.e. if there is an error processing one or more detail record, then we skip the those one and process others. With return it will abort at first failure.

Reply:
Yes, it is changing the behavior and keeping the file on error. I will remove to short the conversation and move on with the PR but I do not recall why we chose this direction. Do you remember? If not, I will start a separate conversation with biz.

Reply:
I believe we chose this direction as each receipt record can be considered independently and processed ignoring one or more records with data errors or other errors. But I would say we should check the direction with business.

Reply:
As we talked, I will check with the business and consider if we should generate the report or not.
For this ticket and this PR, I will keeping the existing behavior as suggested.